pkg/system/phase2_creating.go

```diff
@@ -697,3 +719,57 @@ func (r *Reconciler) UpgradeSplitDBDeleteOldPVC(oldPVC *corev1.PersistentVolumeC
 	}
 	return nil
 }
+
+// UpgradeMigrateDB performs a db upgrade between mongodb to postgres
+func (r *Reconciler) UpgradeMigrateDB() error {
+	phase := r.NooBaa.Status.UpgradePhase
+	if (phase == nbv1.UpgradePhaseFinished) {
+		return nil
+	}
+	oldSts := &appsv1.StatefulSet{
+		TypeMeta: metav1.TypeMeta{Kind: "StatefulSet"},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "noobaa-db",
+			Namespace: options.Namespace,
+		},
+	}
+	if util.KubeCheck(oldSts) {
+		r.Logger.Infof("UpgradeMigrateDB: Old STS found, upgrading...")
+		if (phase == "") {
+			phase = nbv1.UpgradePhasePrepare
+		}
+		switch phase {
+		case nbv1.UpgradePhasePrepare:
+			util.KubeCheckQuiet(r.CoreApp)
+			if (*r.CoreApp.Spec.Replicas == 0) {
+				phase = nbv1.UpgradePhaseMigrate
+			}
+		case nbv1.UpgradePhaseMigrate: 
+			if err:= r.ReconcileObject(r.UpgradeJob, r.SetDesiredJobUpgradeDB); err != nil {
+				return err
+			}
+			if (r.UpgradeJob.Status.Succeeded > 0) {
+				phase = nbv1.UpgradePhaseFinished
```

Comment:
what about cleanup of previous DB?

Reply:
We decided not to delete the pod for now If I remember right. Should I add some kind of scheduled delete? let's discuss.

Reply:
we will set the old sts replicas to 0